CSharpSourceCode/Abilities/AbilityManagerMissionLogic.cs

```diff
@@ -16,16 +16,26 @@ public class AbilityManagerMissionLogic : MissionLogic
         public override void OnMissionTick(float dt)
         {
             base.OnMissionTick(dt);
-            if (Mission.CurrentState == Mission.State.Continuing && Agent.Main != null && Agent.Main.IsAbilityUser())
+
+            if (Mission.IsFriendlyMission) 
+                return;
+
+            if (Mission.CurrentState != Mission.State.Continuing || Agent.Main == null || !Agent.Main.IsAbilityUser()) 
+                return;
+            
+            RegisterInput();
+        }
+
+        private static void RegisterInput()
+        {
+            if (Input.IsKeyPressed(InputKey.Q))
+            {
+                Agent.Main.CastCurrentAbility();
+            }
+
+            if (Input.IsKeyPressed(InputKey.E))
             {
-                if (Input.IsKeyPressed(InputKey.Q))
-                {
-                    Agent.Main.CastCurrentAbility();
-                }
-                if (Input.IsKeyPressed(InputKey.E))
-                {
-                    Agent.Main.SelectNextAbility();
-                }
+                Agent.Main.SelectNextAbility();
             }
         }
     }
```

Comment:
This class has changed in dev since. Your changes are OK, but pls merge the changes from dev your PR first. There was an AgentCreated method added, i would add the IsFriendlyMission check there too.